File: arrays/ScoreofaString.cpp
class Solution {
public:
    int scoreOfString(string s) {
        int score = 0;
        for ( int i = 0; i < s.size() - 1; i++ ) {
            score = score + abs(s[i] - s[i + 1]);
        }
        return score;
    }
};


/**
 * Loop runs with i = 0,1,2,3.

i = 0 → s[0] = 'h' (104), s[1] = 'e' (101)
score = 0 + |104 - 101| = 3

i = 1 → s[1] = 'e' (101), s[2] = 'l' (108)
score = 3 + |101 - 108| = 3 + 7 = 10

i = 2 → s[2] = 'l' (108), s[3] = 'l' (108)
score = 10 + |108 - 108| = 10 + 0 = 10

i = 3 → s[3] = 'l' (108), s[4] = 'o' (111)
score = 10 + |108 - 111| = 10 + 3 = 13
 */

File: arrays/Basic array questions/ScoreofaString.py
"""
Q) score of a string 

You are given a string s. The score of a string is defined as the sum of the absolute difference between the ASCII values of adjacent characters.
Return the score of s.

Example 1:  
Input: s = "abc"
Output: 2

"""

class Solution:
    def scoreOfString(self, s: str) -> int:
        res = 0 

        for i in range(len(s) - 1):
            res += abs(ord(s[i]) - ord(s[i + 1])) #ord = gives the ASCII value of a character, abs = removes negative sign
        return res
        
#explanation of the code
#1. We initialize a variable res to store the score of the string.
#2. We loop through the string from the first character to the second last character.
#3. For each character, we calculate the absolute difference between its ASCII value and the ASCII value of the next character using the ord() function.
#4. We add this absolute difference to res.
#5. Finally, we return the total score stored in res.
#6. The time complexity of this solution is O(n), where n is the length of the string, since we are iterating through the string once. 
#7. The space complexity is O(1) as we are using a constant amount of extra space.